lighthouse-core/test/fraggle-rock/api-test-pptr.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* eslint-env jest */
+
+const path = require('path');
+const lighthouse = require('../../fraggle-rock/api.js');
+const puppeteer = require('puppeteer');
+const StaticServer = require('../../../lighthouse-cli/test/fixtures/static-server.js').Server;
+
+jest.setTimeout(90e3);
+
+describe('Fraggle Rock API', () => {
+  /** @type {InstanceType<StaticServer>} */
+  let server;
+  /** @type {import('puppeteer').Browser} */
+  let browser;
+  /** @type {import('puppeteer').Page} */
+  let page;
+
+  beforeAll(async () => {
+    server = new StaticServer();
+    await server.listen(0, '127.0.0.1');
+    browser = await puppeteer.launch({
+      headless: true,
+    });
+  });
+
+  beforeEach(async () => {
+    page = await browser.newPage();
+  });
+
+  afterEach(async () => {
+    await page.close();
+  });
+
+  afterAll(async () => {
+    await browser.close();
+    await server.close();
+  });
+
+  describe('snapshot', () => {
+    beforeEach(() => {
+      server.setBaseDir(path.join(__dirname, '../fixtures/fraggle-rock/snapshot-basic'));
+    });
+
+    it('should compute accessibility results on the page as-is', async () => {
+      await page.goto(`http://localhost:${server.getPort()}/onclick.html`);
```

Comment:
future tests will slightly benefit if you make a variable for the host ``http://localhost:${server.getPort()}` (in `beforeAll`)

Maybe that'd be better as a property on the server class.

Reply:
I prefer to have it in the test file for trouble of remembering whether it includes the trailing slash ;) but yeah sg I'll pull up to a var

Reply:
same but I've worked around that by adopted a convention that I never ever end a variable with a url string or path component with a slash. ever.